extensions/amp-lightbox-viewer/0.1/amp-lightbox-viewer.js

```diff
@@ -903,14 +904,38 @@ export class AmpLightboxViewer extends AMP.BaseElement {
  */
 export function installLightboxManager(win) {
   if (isExperimentOn(win, TAG)) {
-    // TODO(aghassemi): This only works for singleDoc mode. We will move
+    // TODO (#12859): This only works for singleDoc mode. We will move
     // installation of LightboxManager to core after the experiment, okay for now.
     const ampdoc = Services.ampdocServiceFor(win).getAmpDoc();
     manager_ = new LightboxManager(ampdoc);
   }
 }
 
+/**
+ * Tries to find an existing amp-lightbox-viewer, if there is none, it adds a
+ * default one.
+ * @param {!Window} win
+ * @return {!Promise}
+ */
+function installLightboxViewer(win) {
+  const ampdoc = Services.ampdocServiceFor(win).getAmpDoc();
+  // TODO (#12859): make this work for more than singleDoc mode
+  return ampdoc.whenBodyAvailable().then(body => {
+    const existingViewer = elementByTag(ampdoc.getRootNode(), TAG);
+    if (!existingViewer) {
+      const matches = ampdoc.getRootNode().querySelectorAll('[lightbox]');
```

Comment:
I would install it regardless. elements with `[lightbox]` maybe dynamically loaded through AMP list.